Kernel/KBufferBuilder.h

```diff
@@ -45,6 +45,14 @@ class KBufferBuilder {
     void appendf(const char*, ...);
     void appendvf(const char*, va_list);
 
+    template<typename... Parameters>
+    void appendff(StringView fmtstr, const Parameters&... parameters)
+    {
+        // FIXME: This is really not the way to go about it, but vformat expects a
+        //        StringBuilder. Why does this class exist anyways?
```

Comment:
It exists to build a `KBuffer` instead of a `String`, as `String` is heap-allocated and `kmalloc` heap memory is a scarce resource in kernel land, while `KBuffer` is allocated out of the general VM pool.

Reply:
Both `KBuffer` and `kmalloc` ultimately call `MM.allocate_kernel_region`. ([`kmalloc.cpp:155`][1], [`KBuffer.h:53`][2])

I assume heap memory is a "scarce" resource because it can never be released?

  [1]: https://github.com/SerenityOS/serenity/blob/e49ea1b5209f3b76c9caae317554a6a4a6188d5f/Kernel/Heap/kmalloc.cpp#L155
  [2]: https://github.com/SerenityOS/serenity/blob/e49ea1b5209f3b76c9caae317554a6a4a6188d5f/Kernel/KBuffer.h#L53